src/main/kotlin/org/jmailen/gradle/kotlinter/KotlinterPlugin.kt

```diff
@@ -67,34 +70,34 @@ class KotlinterPlugin : Plugin<Project> {
         }
     }
 
-    private fun Project.registerParentLintTask() =
+    private fun Project.registerParentLintTask(): TaskProvider<Task> =
         tasks.register("lintKotlin") {
             it.group = "formatting"
             it.description = "Runs lint on the Kotlin source files."
         }.also { lintKotlin ->
             tasks.named("check").configure { check -> check.dependsOn(lintKotlin) }
         }
 
-    private fun Project.registerParentFormatTask() =
+    private fun Project.registerParentFormatTask(): TaskProvider<Task> =
         tasks.register("formatKotlin") {
             it.group = "formatting"
             it.description = "Formats the Kotlin source files."
         }
 
-    private fun Project.registerPrePushHookTask() =
+    private fun Project.registerPrePushHookTask(): TaskProvider<InstallPrePushHookTask> =
         tasks.register("installKotlinterPrePushHook", InstallPrePushHookTask::class.java) {
             it.group = "build setup"
             it.description = "Installs Kotlinter Git pre-push hook"
         }
 
-    private fun Project.registerPreCommitHookTask() =
+    private fun Project.registerPreCommitHookTask(): TaskProvider<InstallPreCommitHookTask> =
         tasks.register("installKotlinterPreCommitHook", InstallPreCommitHookTask::class.java) {
             it.group = "build setup"
             it.description = "Installs Kotlinter Git pre-commit hook"
         }
 }
 
-internal val String.id
+internal val String.id: String
     get() = split(" ").first()
 
-internal fun Project.reportFile(name: String) = file("$buildDir/reports/ktlint/$name")
+internal fun Project.reportFile(name: String): File = file("$buildDir/reports/ktlint/$name")
```

Comment:
This one seems obvious without declaring type?

Reply:
Non private functions are suggested to decleared types.